### PR TITLE
Upgrade fluentbit-operator to fluent-operator

### PR DESCRIPTION
**Pull request by wenchajun**
### What this PR does / why we need it:

During the process of upgrading the fluentbit-operator, this script helps the user to convert the original configuration to the new one and to uninstall the `fluentbit-operator`.

### Which issue(s) this PR fixes:

Fixes #

### Does this PR introduced a user-facing change?

```release-note

```

### Additional documentation, usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by benjaminhuo** [comment]
@wanjunlei pls help to review

**Comment by wanjunlei** [comment]
There have two questions：
- The `clusterrole` and `clusterrolebinding` were not deleted.
- There may also be resources in other namespaces, so we cannot only deal with resources under `kubesphere-logging-system`. At the same time, it is necessary to realize that resources under different namespaces may have the same name. This will not be a problem when they are namespace-level resources, but conflicts will occur when they are upgraded to cluster-level resources.

**Review by wanjunlei — CHANGES_REQUESTED** [review]

File: cmd/upgrade/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,11 @@
+FROM alpine:latest
```
[the remark this answers is not in the document]
Comment (benjaminhuo) [reply cmd/upgrade/Dockerfile]: @wanjunlei are u ok with the new change?

[the remark this answers is not in the document]
Comment (wenchajun) [reply cmd/upgrade/Dockerfile]: fixed

[the remark this answers is not in the document]
Comment (benjaminhuo) [reply cmd/upgrade/Dockerfile]: > fixed

@wanjunlei need to take another look

File: cmd/upgrade/upgrade.sh [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+\"name\": ${name[i]},
+\"labels\": ${labels[i]}
+},
+\"spec\": ${spec[i]}
+}"
+fi
+done
+
+## Uninstall the fluentbit-operator and the original configuration
+for((i=0;i<${size};i++));do
```
Comment (wanjunlei) [inline cmd/upgrade/upgrade.sh:46]: Delete after new resource created

**Review by wanjunlei — DISMISSED** [review]

File: cmd/upgrade/upgrade.sh [hunk 3]
```diff
@@ -0,0 +1,80 @@
+#!/bin/bash
+namespace=kubesphere-logging-system
+deployment="fluentbit-operator"
```
Comment (benjaminhuo) [inline cmd/upgrade/upgrade.sh:3]:
```suggestion
fluentbit-operator="fluentbit-operator"
```

File: cmd/upgrade/upgrade.sh [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+converting "inputs" "ClusterInput"
+converting "parsers" "ClusterParser"
+converting "filters" "ClusterFilter"
+converting "outputs" "ClusterOutput"
+converting "fluentbitconfigs" "ClusterFluentBitConfig"
+converting "fluentbits" "FluentBit"
+
+# Determine if Deployment exists
+if kubectl get deployment -n $namespace $deployment >/dev/null 2>&1; then
```
Comment (benjaminhuo) [inline cmd/upgrade/upgrade.sh:67]:
```suggestion
if kubectl get deployment -n $namespace $fluentbit-operator >/dev/null 2>&1; then
```

File: cmd/upgrade/upgrade.sh [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+    # Delete Deployment if it exists
+    kubectl delete deployment -n $namespace $deployment
```
Comment (benjaminhuo) [inline cmd/upgrade/upgrade.sh:69]:
```suggestion
    kubectl delete deployment -n $namespace $fluentbit-operator
```

File: cmd/upgrade/upgrade.sh [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+deployment="fluentbit-operator"
+
+function error_exit {
+  echo "$1" 1>&2
+  exit 1
+}
+
+function converting(){
+## Converting an existing configuration to a new one
+local resource=$1
```
Comment (benjaminhuo) [inline cmd/upgrade/upgrade.sh:12]:
```suggestion
local oldKind=$1
```

File: cmd/upgrade/upgrade.sh [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+local kind=$2
```
Comment (benjaminhuo) [inline cmd/upgrade/upgrade.sh:13]:
```suggestion
local newKind=$2
```

File: cmd/upgrade/upgrade.sh [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+  echo "$temp"
+   kubectl delete  $resource.logging.kubesphere.io $temp -n ${namespace}
+done
+
+for((i=0;i<${size};i++));do
+echo ${cluster_resource_list[i]} | kubectl apply -f - || error_exit "Cannot apply resource $reresource"
+done
+}
+
+converting "inputs" "ClusterInput"
```
Comment (benjaminhuo) [inline cmd/upgrade/upgrade.sh:59]: converting "inputs" "ClusterInput"
converting "Parser" "ClusterParser"
converting "Filter" "ClusterFilter"
converting "Output" "ClusterOutput"
converting "FluentbitConfig" "ClusterFluentBitConfig"
converting "FluentBit" "FluentBit"

File: cmd/upgrade/upgrade.sh (hunk 3 above)
Comment (wenchajun) [reply cmd/upgrade/upgrade.sh]: It has been modified, it is better not to have `-` in the shell name, I modified it to `FluentbitOperator`

File: cmd/upgrade/upgrade.sh [hunk 9]
```diff
@@ -0,0 +1,80 @@
+#!/bin/bash
+namespace=kubesphere-logging-system
+FluentbitOperator="fluentbit-operator"
+
+function error_exit {
+  echo "$1" 1>&2
+  exit 1
+}
+
+function converting(){
```
Comment (benjaminhuo) [inline cmd/upgrade/upgrade.sh:10]: converting can be renamed to migrate

**Review by benjaminhuo — APPROVED** [review]

**Pull request merged**